packages/components/src/local/organisms/admin-login/index.tsx

```diff
@@ -1,50 +1,65 @@
-import React, { useState } from 'react'
-import TextInput from '../../atoms/text-input'
+import { DEFAULT_SERVER, serverPath } from '@serge/config'
+import { RootState } from '@serge/custom-types'
+import React, { useCallback, useMemo, useState } from 'react'
+import { useSelector } from 'react-redux'
 import Button from '../../atoms/button'
-
-/* Import Types */
-import Props from './types/props'
-
-import { DEFAULT_SERVER } from '@serge/config'
-
+import TextInput from '../../atoms/text-input'
 /* Import Stylesheet */
 import styles from './styles.module.scss'
+/* Import Types */
+import Props from './types/props'
 
 /* Render component */
 export const AdminLogin: React.FC<Props> = ({ onSubmit }: Props) => {
   // (temporarily) put password in box, to make testing quicker in dev
   const providePassword = true
   const [password, setPassword] = useState(providePassword ? DEFAULT_SERVER : '')
+  const gameInfo = useSelector((state: RootState) => state.gameInfo)
 
-  const setPasswordValue = (target: { value: string }): void => {
+  const setPasswordValue = useCallback((target: { value: string }): void => {
     setPassword(target.value)
-  }
-  const handleOnKeyPress = (evt: React.KeyboardEvent): void => {
+  }, [])
+
+  const handleOnKeyPress = useCallback((evt: React.KeyboardEvent): void => {
     if (evt.key === 'Enter') {
       onSubmit && onSubmit(password)
     }
-  }
+  }, [password])
+
+  const logoPath = useMemo(() => {
+    if (gameInfo.imageUrl) {
+      return `${serverPath.substring(0, serverPath.length - 1)}${gameInfo.imageUrl}`
+    }
+    return ''
+  }, [gameInfo.imageUrl])
 
   return (
     <>
-      <h2 className={styles['login-form-head']}>Password</h2>
-      <div style={{ marginBottom: 16 }}>
-        <TextInput
-          label="Password"
-          labelColor="common.white"
-          value={password}
-          type="password"
-          variant="filled"
-          updateState={setPasswordValue}
-          onKeyPress={handleOnKeyPress}
-          fullWidth
-        />
-      </div>
-      <Button color="secondary" onClick={(): void => onSubmit(password)}>
+      <div className={styles.container}>
+        <span className={styles.logo}>
+          <img src={logoPath} />
+          <p>Serge</p>
+          <p>Admin Page</p>
+        </span>
+
+        <div className={styles.password}>
+          <TextInput
+            label="Password"
+            labelColor="common.white"
+            value={password}
+            type="password"
+            variant="filled"
+            updateState={setPasswordValue}
+            onKeyPress={handleOnKeyPress}
+            fullWidth
+          />
+        </div>
+        <Button color="primary" onClick={(): void => onSubmit(password)}>
         Enter
-      </Button>
+        </Button>
+      </div>
     </>
   )
 }
 
-export default AdminLogin
+export default React.memo(AdminLogin)
```

Comment:
the `admin-login` component has the callback function `onSubmit`, so we should apply the memoization. You can see here https://dmitripavlutin.com/use-react-memo-wisely/#:~:text=memo()%20wisely.,memoization%20for%20class%2Dbased%20components

Reply:
Another one @IanMayo 🤣  https://javascript.plainenglish.io/when-and-when-not-to-use-react-memo-e050d3cb1ffc
<img width="714" alt="image" src="https://user-images.githubusercontent.com/26705621/158167336-ba71b440-da48-4bd2-82da-85b970ae15fb.png">

Reply:
```
When we have pure functional components. Given the same props, such components will remain the same — always rendering the same output.
```
In case if the state does not changed, using `React.memo()` will help use to avoid the component re-render often I think. So I think it's a good habit.

Reply:
Sure.  But the `admin page` doesn't re-render a lot.  Hmm, I don't think it even takes any props.  If it does, they won't change.

So, this _feels like_ we're adding complexity without justification.

Reply:
yeah, so we can remove it 😄